plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Definition.java

```diff
@@ -351,16 +355,32 @@ private Transform(final Cast cast, Method method, final Type upcast, final Type
     final Type utilityType;
     final Type defobjType;
 
+    final Type itrType;
+    final Type oitrType;
+    final Type sitrType;
+
+    final Type collectionType;
+    final Type ocollectionType;
+    final Type scollectionType;
+
     final Type listType;
     final Type arraylistType;
-    final Type mapType;
-    final Type hashmapType;
-
     final Type olistType;
     final Type oarraylistType;
-    final Type omapType;
-    final Type ohashmapType;
+    final Type slistType;
```

Comment:
In this case, it's not possible.  The types haven't been created until the constructor is called.